### ISSUE TITLE
no ring while typing with sway 1.6-1 / libwlroots8 0.13.0-1

### ISSUE THREAD

**Issue by 532910** [swaywm/swaylock#183]
1. Under sway 1.5.1-2 / libwlroots7 0.12.0-2 swaylock works fine.
2. Under sway 1.6-1 / libwlroots8 0.13.0-1 there are no ring while typing.
3. `Cleared` ring is shown with both sway versions.
4. With `--indicator-radius=100` it works fine under sway 1.6-1.

Debian sid/experimental.

**Comment by emersion** [comment]
Is this the same as https://github.com/swaywm/swaylock/issues/166? Are you seeing "Buffer size not divisible by scale" errors in the Sway debug log?

**Comment by 532910** [comment]
There no errors at all:
```
% swaylock -d -C /dev/null --color=000000                                                                                                                     
2021-05-27 11:52:11 - [main.c:1137] Found config at /dev/null
2021-05-27 11:52:11 - [main.c:1147] Parsing CLI Args
2021-05-27 11:52:11 - [main.c:280] output name is eDP-1
```
<details>
<summary> % WAYLAND_DEBUG=1 swaylock -d -C /dev/null --color=000000 </summary>

```
2021-05-27 11:56:09 - [main.c:1137] Found config at /dev/null
2021-05-27 11:56:09 - [main.c:1147] Parsing CLI Args
[3996335.647]  -> wl_display@1.get_registry(new id wl_registry@2)
[3996335.729]  -> wl_display@1.sync(new id wl_callback@3)
[3996335.991] wl_display@1.delete_id(3)
[3996336.055] wl_registry@2.global(1, "wl_shm", 1)
[3996336.110]  -> wl_registry@2.bind(1, "wl_shm", 1, new id [unknown]@4)
[3996336.179] wl_registry@2.global(2, "wl_drm", 2)
[3996336.226] wl_registry@2.global(3, "zwp_linux_dmabuf_v1", 3)
[3996336.271] wl_registry@2.global(4, "wl_compositor", 4)
[3996336.316]  -> wl_registry@2.bind(4, "wl_compositor", 3, new id [unknown]@5)
[3996336.380] wl_registry@2.global(5, "wl_subcompositor", 1)
[3996336.425]  -> wl_registry@2.bind(5, "wl_subcompositor", 1, new id [unknown]@6)
[3996336.488] wl_registry@2.global(6, "wl_data_device_manager", 3)
[3996336.534] wl_registry@2.global(7, "zwlr_gamma_control_manager_v1", 1)
[3996336.579] wl_registry@2.global(8, "zxdg_output_manager_v1", 3)
[3996336.625]  -> wl_registry@2.bind(8, "zxdg_output_manager_v1", 2, new id [unknown]@7)
[3996336.687] wl_registry@2.global(9, "org_kde_kwin_idle", 1)
[3996336.732] wl_registry@2.global(10, "zwp_idle_inhibit_manager_v1", 1)
[3996336.777] wl_registry@2.global(11, "zwlr_layer_shell_v1", 4)
[3996336.823]  -> wl_registry@2.bind(11, "zwlr_layer_shell_v1", 1, new id [unknown]@8)
[3996336.886] wl_registry@2.global(12, "xdg_wm_base", 2)
[3996336.931] wl_registry@2.global(13, "zwp_tablet_manager_v2", 1)
[3996336.976] wl_registry@2.global(14, "org_kde_kwin_server_decoration_manager", 1)
[3996337.021] wl_registry@2.global(15, "zxdg_decoration_manager_v1", 1)
[3996337.066] wl_registry@2.global(16, "zwp_relative_pointer_manager_v1", 1)
[3996337.111] wl_registry@2.global(17, "zwp_pointer_constraints_v1", 1)
[3996337.155] wl_registry@2.global(18, "wp_presentation", 1)
[3996337.200] wl_registry@2.global(19, "zwlr_output_manager_v1", 2)
[3996337.245] wl_registry@2.global(20, "zwlr_output_power_manager_v1", 1)
[3996337.290] wl_registry@2.global(21, "zwp_input_method_manager_v2", 1)
[3996337.334] wl_registry@2.global(22, "zwp_text_input_manager_v3", 1)
[3996337.379] wl_registry@2.global(23, "zwlr_foreign_toplevel_manager_v1", 3)
[3996337.423] wl_registry@2.global(24, "zwlr_export_dmabuf_manager_v1", 1)
[3996337.513] wl_registry@2.global(25, "zwlr_screencopy_manager_v1", 3)
[3996337.561] wl_registry@2.global(26, "zwlr_data_control_manager_v1", 2)
[3996337.605] wl_registry@2.global(27, "zwp_primary_selection_device_manager_v1", 1)
[3996337.651] wl_registry@2.global(28, "wp_viewporter", 1)
[3996337.696] wl_registry@2.global(29, "zxdg_exporter_v1", 1)
[3996337.741] wl_registry@2.global(30, "zxdg_importer_v1", 1)
[3996337.786] wl_registry@2.global(31, "zxdg_exporter_v2", 1)
[3996337.831] wl_registry@2.global(32, "zxdg_importer_v2", 1)
[3996337.876] wl_registry@2.global(33, "zwp_virtual_keyboard_manager_v1", 1)
[3996337.921] wl_registry@2.global(34, "zwlr_virtual_pointer_manager_v1", 2)
[3996337.966] wl_registry@2.global(35, "zwlr_input_inhibit_manager_v1", 1)
[3996338.013]  -> wl_registry@2.bind(35, "zwlr_input_inhibit_manager_v1", 1, new id [unknown]@9)
[3996338.076] wl_registry@2.global(36, "zwp_keyboard_shortcuts_inhibit_manager_v1", 1)
[3996338.122] wl_registry@2.global(37, "wl_seat", 7)
[3996338.167]  -> wl_registry@2.bind(37, "wl_seat", 3, new id [unknown]@10)
[3996338.231] wl_registry@2.global(38, "zwp_pointer_gestures_v1", 1)
[3996338.276] wl_registry@2.global(39, "wl_output", 3)
[3996338.322]  -> wl_registry@2.bind(39, "wl_output", 3, new id [unknown]@11)
[3996338.391] wl_callback@3.done(1213)
[3996338.428]  -> zwlr_input_inhibit_manager_v1@9.get_inhibitor(new id zwlr_input_inhibitor_v1@3)
[3996338.471]  -> wl_display@1.sync(new id wl_callback@12)
[3996339.628] wl_display@1.delete_id(12)
[3996339.700] wl_seat@10.name("seat0")
[3996339.725] wl_seat@10.capabilities(3)
[3996339.753]  -> wl_seat@10.get_pointer(new id wl_pointer@13)
[3996339.785]  -> wl_seat@10.get_keyboard(new id wl_keyboard@14)
[3996339.815] wl_output@11.geometry(0, 0, 350, 190, 0, "Unknown", "0x06D7", 0)
[3996339.961] wl_output@11.mode(1, 3840, 2160, 59997)
[3996340.025] wl_output@11.scale(2)
[3996340.051] wl_output@11.done()
[3996340.069] wl_callback@12.done(1216)
[3996340.100]  -> zxdg_output_manager_v1@7.get_xdg_output(new id zxdg_output_v1@12, wl_output@11)
[3996340.154]  -> wl_display@1.sync(new id wl_callback@15)
[3996340.762] wl_display@1.delete_id(15)
[3996340.831] wl_keyboard@14.keymap(1, fd 6, 53228)
[3996346.968] zxdg_output_v1@12.name("eDP-1")
2021-05-27 11:56:09 - [main.c:280] output name is eDP-1
[3996347.048] zxdg_output_v1@12.description("Unknown 0x06D7 0x00000000 (eDP-1)")
[3996347.065] zxdg_output_v1@12.logical_position(0, 0)
[3996347.089] zxdg_output_v1@12.logical_size(1920, 1080)
[3996347.113] zxdg_output_v1@12.done()
[3996347.124] wl_output@11.done()
[3996347.135] wl_callback@15.done(1216)
[3996347.167]  -> wl_compositor@5.create_surface(new id wl_surface@15)
[3996347.191]  -> wl_compositor@5.create_surface(new id wl_surface@16)
[3996347.211]  -> wl_subcompositor@6.get_subsurface(new id wl_subsurface@17, wl_surface@16, wl_surface@15)
[3996347.247]  -> wl_subsurface@17.set_sync()
[3996347.261]  -> zwlr_layer_shell_v1@8.get_layer_surface(new id zwlr_layer_surface_v1@18, wl_surface@15, wl_output@11, 3, "lockscreen")
[3996347.313]  -> zwlr_layer_surface_v1@18.set_size(0, 0)
[3996347.337]  -> zwlr_layer_surface_v1@18.set_anchor(15)
[3996347.352]  -> zwlr_layer_surface_v1@18.set_exclusive_zone(-1)
[3996347.367]  -> zwlr_layer_surface_v1@18.set_keyboard_interactivity(1)
[3996347.382]  -> wl_compositor@5.create_region(new id wl_region@19)
[3996347.402]  -> wl_region@19.add(0, 0, 2147483647, 2147483647)
[3996347.442]  -> wl_surface@15.set_opaque_region(wl_region@19)
[3996347.458]  -> wl_region@19.destroy()
[3996347.470]  -> wl_surface@15.commit()
[3996347.661] wl_display@1.delete_id(19)
[3996347.695] zwlr_layer_surface_v1@18.configure(1217, 1920, 1080)
[3996347.727]  -> zwlr_layer_surface_v1@18.ack_configure(1217)
[3996347.835]  -> wl_shm@4.create_pool(new id wl_shm_pool@19, fd 7, 33177600)
[3996347.892]  -> wl_shm_pool@19.create_buffer(new id wl_buffer@20, 0, 3840, 2160, 15360, 0)
[3996347.951]  -> wl_shm_pool@19.destroy()
[3996371.717]  -> wl_surface@15.set_buffer_scale(2)
[3996371.744]  -> wl_surface@15.attach(wl_buffer@20, 0, 0)
[3996371.758]  -> wl_surface@15.damage(0, 0, 1920, 1080)
[3996371.772]  -> wl_surface@15.commit()
[3996371.777]  -> wl_subsurface@17.set_position(961, 480)
[3996371.831]  -> wl_shm@4.create_pool(new id wl_shm_pool@21, fd 8, 4)
[3996371.847]  -> wl_shm_pool@21.create_buffer(new id wl_buffer@22, 0, 1, 1, 4, 0)
[3996371.869]  -> wl_shm_pool@21.destroy()
[3996371.892]  -> wl_surface@16.attach(nil, 0, 0)
[3996371.905]  -> wl_surface@16.commit()
[3996371.919]  -> wl_surface@16.set_buffer_scale(2)
[3996371.926]  -> wl_surface@16.attach(wl_buffer@22, 0, 0)
[3996371.937]  -> wl_surface@16.damage(0, 0, 1, 1)
[3996371.952]  -> wl_surface@16.commit()
[3996371.956]  -> wl_surface@15.commit()
[3996402.800] wl_display@1.delete_id(19)
[3996402.832] wl_display@1.delete_id(21)
[3996402.838] wl_buffer@20.release()
[3996402.843] wl_pointer@13.enter(1218, wl_surface@15, 775.203125, 196.417969)
[3996402.858]  -> wl_pointer@13.set_cursor(1218, nil, 0, 0)
[3996402.872] wl_keyboard@14.enter(1219, wl_surface@15, array)
[3996402.883] wl_keyboard@14.modifiers(1220, 0, 0, 0, 0)
[3996448.011] wl_keyboard@14.key(1221, 23140767, 28, 0)
[3997376.505] wl_keyboard@14.key(1222, 23141695, 36, 1)
[3997376.544]  -> wl_surface@15.frame(new id wl_callback@21)
[3997376.554]  -> wl_surface@15.commit()
[3997378.239] wl_display@1.delete_id(21)
[3997378.262] wl_callback@21.done(23141697)
[3997378.270]  -> wl_surface@15.frame(new id wl_callback@21)
[3997378.278]  -> wl_subsurface@17.set_position(961, 480)
[3997378.330]  -> wl_shm@4.create_pool(new id wl_shm_pool@19, fd 7, 4)
[3997378.353]  -> wl_shm_pool@19.create_buffer(new id wl_buffer@23, 0, 1, 1, 4, 0)
[3997378.377]  -> wl_shm_pool@19.destroy()
[3997378.396]  -> wl_surface@16.attach(nil, 0, 0)
[3997378.409]  -> wl_surface@16.commit()
[3997388.993]  -> wl_buffer@23.destroy()
[3997389.027]  -> wl_subsurface@17.set_position(901, 480)
[3997389.067]  -> wl_shm@4.create_pool(new id wl_shm_pool@24, fd 8, 283200)
[3997389.085]  -> wl_shm_pool@24.create_buffer(new id wl_buffer@25, 0, 240, 295, 960, 0)
[3997389.107]  -> wl_shm_pool@24.destroy()
[3997389.121]  -> wl_surface@16.attach(nil, 0, 0)
[3997389.134]  -> wl_surface@16.commit()
[3997390.223]  -> wl_surface@16.set_buffer_scale(2)
[3997390.235]  -> wl_surface@16.attach(wl_buffer@25, 0, 0)
[3997390.248]  -> wl_surface@16.damage(0, 0, 240, 295)
[3997390.262]  -> wl_surface@16.commit()
[3997390.266]  -> wl_surface@15.commit()
[3997390.270]  -> wl_surface@16.set_buffer_scale(2)
[3997390.276]  -> wl_surface@16.attach(wl_buffer@25, 0, 0)
[3997390.287]  -> wl_surface@16.damage(0, 0, 240, 295)
[3997390.301]  -> wl_surface@16.commit()
[3997390.305]  -> wl_surface@15.commit()
[3997399.964] wl_display@1.delete_id(19)
[3997399.991] wl_display@1.delete_id(23)
[3997399.998] wl_display@1.delete_id(24)
[3997400.004] wl_display@1.delete_id(21)
[3997400.010] wl_callback@21.done(23141719)
[3997438.051] wl_keyboard@14.key(1223, 23141757, 23, 1)
[3997438.175]  -> wl_surface@15.frame(new id wl_callback@21)
[3997438.210]  -> wl_surface@15.commit()
[3997438.746] wl_display@1.delete_id(21)
[3997438.809] wl_callback@21.done(23141758)
[3997438.838]  -> wl_surface@15.frame(new id wl_callback@21)
[3997438.871]  -> wl_subsurface@17.set_position(901, 480)
[3997480.352] wl_keyboard@14.key(1224, 23141799, 36, 0)
[3997565.443] wl_keyboard@14.key(1225, 23141884, 23, 0)
[3997565.656] wl_keyboard@14.key(1226, 23141884, 18, 1)
[3997634.973] wl_keyboard@14.key(1227, 23141954, 47, 1)
[3997703.972] wl_keyboard@14.key(1228, 23142023, 18, 0)
[3997720.861] wl_keyboard@14.key(1229, 23142040, 7, 1)
[3997773.391] wl_keyboard@14.key(1230, 23142092, 47, 0)
[3997798.483] wl_keyboard@14.key(1231, 23142117, 7, 0)
[3997843.478] wl_keyboard@14.key(1232, 23142162, 42, 1)
[3997843.535] wl_keyboard@14.modifiers(1233, 1, 0, 0, 0)
[3997936.187] wl_keyboard@14.key(1234, 23142255, 38, 1)
[3998001.634] wl_keyboard@14.key(1235, 23142320, 42, 0)
[3998001.806] wl_keyboard@14.modifiers(1236, 0, 0, 0, 0)
[3998091.979] wl_keyboard@14.key(1237, 23142411, 38, 0)
[3998092.162] wl_keyboard@14.key(1238, 23142411, 30, 1)
[3998124.111] wl_keyboard@14.key(1239, 23142443, 35, 1)
[3998174.542] wl_keyboard@14.key(1240, 23142493, 30, 0)
[3998201.516] wl_keyboard@14.key(1241, 23142520, 35, 0)
[3998282.429] wl_keyboard@14.key(1242, 23142601, 28, 1)
```
</details>

**Comment by emersion** [comment]
I meant Sway debug logs, not swyalock debug logs.

**Comment by 532910** [comment]
Still nothing:

```
% WAYLAND_DEBUG=1 =sway >& /tmp/log
% grep 'Buffer size not divisible by scale' /tmp/log 
zsh: exit 1     grep --color=always --initial-tab 'Buffer size not divisible by scale'
% grep 'divisible' /tmp/log     
zsh: exit 1     grep --color=always --initial-tab 'divisible' /tmp/log
```

**Comment by emersion** [comment]
Can you confirm Sway was started with `-d`?

**Comment by 532910** [comment]
Nope, but I just re-done it with with `-d` and still see no `Buffer size not divisible by scale`. Both sway and swaylock was started with `-d` the last time.

**Comment by emersion** [comment]
Can you still reproduce with https://github.com/swaywm/swaylock/pull/187?

**Comment by 532910** [comment]
Master works fine, but now it shows layout (English) only after a second key press, should I open a new issue?

**Comment by emersion** [comment]
Please do, seems separate.
